pkg/efscreate/create.go

```diff
@@ -0,0 +1,171 @@
+package efscreate
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"os"
+	"strings"
+)
+
+const (
+	operatorName          = "create-efs-volume"
+	infraGlobalName       = "cluster"
+	secretNamespace       = "kube-system"
+	secretName            = "aws-creds"
+	STORAGECLASS_LOCATION = "STORAGEClASS_LOCATION"
+	MANIFEST_LOCATION     = "MANIFEST_LOCATION"
+)
+
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	// Create core clientset for core and infra objects
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	nodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
```

Comment:
The API server (or network or load balancer) tends to fail, especially shortly after cluster installation. There should be generous retry in all API server interactions (list nodes, get infrastructure, get AWS secret).